oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -111,28 +123,36 @@ private void createNormalTable(Model model) throws StorageException {
         ElasticSearchClient esClient = (ElasticSearchClient) client;
         String tableName = IndexController.INSTANCE.getTableName(model);
         Mappings mapping = createMapping(model);
+        Map<String, Object> settings = createSetting(model);
         if (!esClient.isExistsIndex(tableName)) {
-            Map<String, Object> settings = createSetting(model);
             boolean isAcknowledged = esClient.createIndex(tableName, mapping, settings);
             log.info("create {} index finished, isAcknowledged: {}", tableName, isAcknowledged);
             if (!isAcknowledged) {
-                throw new StorageException("create " + tableName + " index failure, ");
+                throw new StorageException("create " + tableName + " index failure");
             }
         } else {
             Mappings historyMapping = esClient.getIndex(tableName)
                                               .map(Index::getMappings)
                                               .orElseGet(Mappings::new);
-            structures.putStructure(tableName, mapping);
-            Mappings appendMapping = structures.diffStructure(tableName, historyMapping);
+            structures.putStructure(tableName, mapping, settings);
+            Mappings appendMapping = structures.diffMappings(tableName, historyMapping);
+            //update mapping
             if (appendMapping.getProperties() != null && !appendMapping.getProperties().isEmpty()) {
                 boolean isAcknowledged = esClient.updateIndexMapping(tableName, appendMapping);
-                log.info("update {} index finished, isAcknowledged: {}, append mappings: {}", tableName,
+                log.info("update {} index mapping finished, isAcknowledged: {}, append mapping: {}", tableName,
                          isAcknowledged, appendMapping
                 );
                 if (!isAcknowledged) {
-                    throw new StorageException("update " + tableName + " index failure");
+                    throw new StorageException("update " + tableName + " index mapping failure");
                 }
             }
+            //needs to update settings
+            if (!structures.compareIndexSetting(tableName, settings)) {
+                log.warn(
+                    "index {} settings configuration has been updated to {}, please remove it before OAP starts",
+                    tableName, settings
+                );
+            }
```

Comment:
Under which scenario, which type of setting, causes that the index needs to be removed? I guess that if specific number of shards or replicas are specified for a concrete index, this warning will be displayed isn't it?
I not sure if we should display it when only shards or replicas have changed:

1. User could be confused, as the shards and replica changes only apply for new indices after the changes are applied to the index template as per https://github.com/apache/skywalking/pull/9914/files#diff-7c42c99e33896647d67385263076ec5178df8e57d072e760895ed7ca9f5d1c20R199
2. Maybe we should display an info message instead if only shards and replica number have changed, as those changes can never be applied because ES doesn't allow to change the number of shards and replicas for and existing index. Moreover, I think those changes are never applied to an existing index, that's the reason why I would suggest to not show a warning message.

Reply:
Hi, this warning is only for the `Normal index` which would not generate new indexes with time series by index template. So the new settings will not be applied until the index is recreated. 

For the `Time series index` I log the info:
https://github.com/apache/skywalking/blob/48a0985345bc65a63b17d4539f80de334383dedb/oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java#L199